src/fromager/resolver.py

```diff
@@ -184,13 +186,21 @@ def find_matches(self, identifier, requirements, incompatibilities):
         # Need to pass the extras to the search, so they
         # are added to the candidate at creation - we
         # treat candidates as immutable once created.
-        candidates = (
-            candidate
-            for candidate in get_project_from_pypi(identifier, set(), self.sdist_server_url)
-            if (candidate.is_sdist or not self.only_sdists)
-            and candidate.version not in bad_versions
-            and all(candidate.version in r.specifier for r in requirements)
-        )
+        candidates = []
+        for candidate in get_project_from_pypi(identifier, set(), self.sdist_server_url):
+            # Skip versions that are known bad
+            if candidate.version in bad_versions:
+                continue
+            # Skip versions that do not match the requirement
+            if not all(candidate.version in r.specifier for r in requirements):
+                continue
+            # Only include sdists if we're asked to
+            if self.include_sdists and not candidate.is_sdist:
+                continue
+            # Only include wheels if we're asked to
+            if self.include_wheels and candidate.is_sdist:
```

Comment:
Are there other things on PYPI that might be neither wheel nor sdist?

Reply:
There's an old format called "egg" but it's obsolete so if we did find any we wouldn't want to use them.